src/config.cpp

```diff
@@ -969,6 +969,10 @@ Config::Config(const fs::path& path, std::string_view json_overlay) : config_pat
   for (const auto& provider_option : model.decoder.session_options.provider_options) {
     model.decoder.session_options.providers.push_back(provider_option.name);
   }
+
+  for (const auto& provider_option : model.encoder.session_options.provider_options) {
+    model.encoder.session_options.providers.push_back(provider_option.name);
+  }
```

Comment:
Should the encoder providers be made configurable at runtime? Or is the purpose for this PR to only allow the encoder session options to be read from the genai_config specifically for the whisper model?

Reply:
Hi @baijumeswani -- I think eventually it would be good to expose some APIs that would allow the encoder providers to be configured at runtime. But for this PR, the intention is allow the session options to get read from `genai_config.json`, specifically.